core/server/ntske.go

```diff
@@ -0,0 +1,102 @@
+package server
+
+import (
+	"context"
+	"crypto/tls"
+	"net"
+	"strconv"
+
+	"go.uber.org/zap"
+
+	"example.com/scion-time/net/ntske"
+)
+
+const defaultNtskePort int = 4460
+
+func handleKeyExchange(log *zap.Logger, ke *ntske.KeyExchange, localHost *net.UDPAddr, provider *ntske.Provider) {
+	defer ke.Conn.Close()
+
+	err := ke.Read()
+	if err != nil {
+		log.Info("failed to read key exchange", zap.Error(err))
+		return
+	}
+
+	err = ke.ExportKeys()
+	if err != nil {
+		log.Info("failed to export keys", zap.Error(err))
+		return
+	}
+
+	var msg ntske.ExchangeMsg
+	msg.AddRecord(ntske.NextProto{
+		NextProto: ntske.NTPv4,
+	})
+	msg.AddRecord(ntske.Algorithm{
+		Algo: []uint16{ntske.AES_SIV_CMAC_256},
+	})
+	msg.AddRecord(ntske.Server{
+		Addr: []byte(localHost.IP.String()),
+	})
+	msg.AddRecord(ntske.Port{
+		Port: uint16(localHost.Port),
+	})
+
+	var plaintextCookie ntske.ServerCookie
+	plaintextCookie.Algo = ntske.AES_SIV_CMAC_256
+	plaintextCookie.C2S = ke.Meta.C2sKey
+	plaintextCookie.S2C = ke.Meta.S2cKey
+	key := provider.Current()
+	for i := 0; i < 8; i++ {
+		encryptedCookie, err := plaintextCookie.EncryptWithNonce(key.Value, key.Id)
+		if err != nil {
+			log.Info("failed to encrypt cookie", zap.Error(err))
+			continue
+		}
+
+		b := encryptedCookie.Encode()
+		msg.AddRecord(ntske.Cookie{
+			Cookie: b,
+		})
+	}
```

Comment:
Do we need the check for at least one cookie added to the response also here?

Reply:
Done